### ISSUE TITLE
Can't update the device type via inventory.json

### ISSUE THREAD

**Issue by anonymosh** [thin-edge/thin-edge.io#1935]
**Describe the bug**
We are trying to set the device type in the inventory.json of thin-edge.io alongside other fragments. E.g.
`{"type": "PumpAgent", "name": "inventoryname", "someothefragment":{}}`

**Expected behavior**
The expected behaviour is that the type behaves here like all the other fragments on the device, i.e. that the type info from the inventory.json is used for the ManagedObject in Cumulocity.

**Observed behavior**
The _type_ is ignored, instead it looks like the tedge config for device.type is always used instead. 
As the inventory.json is more specific, I would have expected the entry there to take precedence

**Screenshots**
![image](https://user-images.githubusercontent.com/18506071/235160351-62dd6c4a-568e-4770-8213-0c75b4659698.png)

**Environment (please complete the following information):**
- Ubuntu 22.04
 - thin-edge.io version [e.g. 0.10.0]

Thanks,
Mario

**Comment by reubenmiller** [comment]
Thanks @anonymosh for creating the ticket. I wouldn't necessarily class it as a bug, but I can understand the use-case as keeping all of the inventory settings in one file is more convenient rather than setting some setting via `tedge.toml` and via `inventory.json`.

@anonymosh In the event that the device type is set in both the `tedge.toml` and the `inventory.json`, then is it ok if the `inventory.json` value is used?

**Comment by anonymosh** [comment]
Yep, I think that would make the behavior more consistent. ( that is inventory.json being more specific and therefore taking preference)

**Comment by reubenmiller** [comment]
Great news, in the future 1.5.0 release, you will finally be able to set the `type` and `name` properties in the `inventory.json` file!

But you can try it out today by using our "main" release channel if you don't want to wait for 1.5.0.

```sh
wget -O - thin-edge.io/install.sh | sh -s -- --channel main
```

Then you can edit the `/etc/tedge/device/inventory.json` file with any properties you want, and it will be sent to Cumulocity each time the `tedge-mapper-c8y` starts up.

Below is an example of the `/etc/tedge/device/inventory.json` file.

```sh
{
 "name": "myCustomName",
 "type": "fancy-new-ai-chip",
 "foo": {
  "name":"bar"
  }
}
```

The inventory.json file is more for static data (e.g. something that does not really change after the device starts up, e.g. operating system version etc.)

Internally, the inventory.json file is just being published as individual MQTT retained messages to the `te/{topic-id}/twin/{type}` topic. So this allows other applications to subscribe/read the information.

Below shows an example of subscribing to the `/twin/+` topic (using a MQTT wildcard, `+`)

```
$ tedge mqtt sub 'te/device/main///twin/+'

[te/device/main///twin/name] "myCustomName"
[te/device/main///twin/type] "fancy-new-ai-chip"
[te/device/main///twin/foo] {"name":"bar"}
```

If you need something more dynamic, then you can also just publish directly to the `/twin/{type}` topics. Though the Community plugin, [tedge-inventory-plugin](https://github.com/thin-edge/tedge-inventory-plugin) might be of interest to you, it also includes some default scripts which will publish the device's OS information and location information
